data-prepper-plugins/geoip-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/MaxMindGeoIpProvider.java

```diff
@@ -43,11 +61,88 @@ public MaxMindGeoIpProvider(String databasePath) {
         try {
             final InetAddress ipAddress = InetAddress.getByName(IpAddress);
             final CityResponse response = databaseReader.city(ipAddress);
-            //TODO change what kind of data is returned, allowing users to submit desired fields, etc.
             final Country country = response.getCountry();
             final Subdivision subdivision = response.getMostSpecificSubdivision();
             final City city = response.getCity();
-            final LocationData returnData = new LocationData(country.getName(), subdivision.getName(), city.getName());
+            final Location location = response.getLocation();
+            final Continent continent = response.getContinent();
+            final Postal postal = response.getPostal();
+            final LocationData.Builder builder = new LocationData.Builder();
+            for (GeoDataField desiredField : this.fieldsToAdd) {
+                switch (desiredField) {
+                    case IP:
+                        builder.withIp(ipAddress.getHostAddress());
+                        break;
+                    case CITY_NAME:
+                        String cityName = city.getName();
+                        if (cityName != null) {
+                            builder.withCity(cityName);
+                        }
```

Comment:
Do we need the null checks? e.g. `builder.WithCity(cityName);`  (also suggest the builder method be `withCityName`)